### PR TITLE
Configuration to avoid fetching resource

### PR DESCRIPTION
**Pull request by afallou**
**Why?**
This is part of POC to transition to handlers that don't fetch the published resource, and instead only use the message itself.

**What?**
- Add configuration value to bypass fetching the resource in URIHandler
- Add base handler that has the same behavior, and that will be used instead of URIHandler in the future.

### CODE REVIEW

File: microcosm_pubsub/handlers/base.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        # Nb. likely to happen with mocks and tests.
+        return False
+
+    return all((
+        search(r"/[a-z]+_event/", uri),
+        search(r".{}.".format(LifecycleChange.Created), media_type),
+    ))
+
+
+class PubSubHandler(metaclass=ABCMeta):
```
Comment (afallou) [inline microcosm_pubsub/handlers/base.py]: Everything in this class was taken from `URIHandler`.

File: microcosm_pubsub/handlers/uri_handler.py [hunk 2]
```diff
@@ -42,121 +17,33 @@ class URIHandler(metaclass=ABCMeta):
...
 
-        resource = self.convert_resource(
-            self.get_resource(message, uri),
-        )
+        resource = None
+        # XXX: remove conditional and use base class instead post-POC
+        if self.fetch_resource:
+            resource = self.convert_resource(
+                self.get_resource(message, uri),
+            )
```
Comment (afallou) [inline microcosm_pubsub/handlers/uri_handler.py]: Those lines are the only difference with the base class for the `__call__` method.

File: microcosm_pubsub/handlers/base.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        resource_cache_whitelist_callable=resource_cache_whitelist_callable,
+    ):
+        self.opaque = graph.opaque
+        self.retry_nack_timeout = retry_nack_timeout
+        self.resource_nack_timeout = resource_nack_timeout
+        self.resource_cache_ttl = resource_cache_ttl
+        self.resource_cache = self.get_resource_cache(graph) if resource_cache_enabled else None
+        self.resource_cache_whitelist_callable = resource_cache_whitelist_callable
+
+    @property
```
Comment (GabeLambda) [inline microcosm_pubsub/handlers/base.py]: We can probably just remove this if we're going all-new here.

Comment (afallou) [reply microcosm_pubsub/handlers/base.py]: I'll move it to `URIHandler` instead of the base class.

**Pull request merged**